dbms/src/Storages/DeltaMerge/ReadThread/CircularScanList.h

```diff
@@ -97,6 +96,12 @@ class CircularScanList
         }
     }
 
+    bool needScheduled(Iter itr)
+    {
+        // If other components hold this SegmentReadTaskPool, schedule it for read blocks or clean MergedTaskPool if necessary.
+        return (*itr)->valid() || itr->use_count() > 1;
+    }
```

Comment:
Will a shared pointer of SegmentReadTaskPool only be held in `SegmentReadTaskScheduler::SegmentReadTaskPoolList` (so that the use_count() == 1) with valid() == false?